libs/GameKit/source/games/GameSpeed.cpp

```diff
@@ -0,0 +1,112 @@
+
+
+#include "games/GameSpeed.h"
+
+#include "drivers/HighResClock.h"
+#include "rtos/ThisThread.h"
+
+using namespace leka;
+using namespace std::chrono;
+namespace {
+void do_nothing(leka::Position position) {}
+}	// namespace
+
+void GameSpeed::start()
+{
+	choseRandomCaptorAsTarget();
+
+	_startTime				 = rtos::Kernel::Clock::now();
+	auto update_touched_func = [&](Position position) { updateTouchedColor(position); };
+	_touch_sensor_kit.init();
+	_touch_sensor_kit.registerOnSensorTouched(update_touched_func);
+
+	_touch_sensor_kit.start();
+	_running = true;
+}
+
+void GameSpeed::run()
+{
+	if (_score == _scoreToWinLevel) {
+		_touch_sensor_kit.registerOnSensorTouched(do_nothing);
+		_ledManager.playReinforcer();
+
+		_running = false;
+
+	} else if (_win || !checkForTimeout()) {
+		choseRandomCaptorAsTarget();
+		_startTime = rtos::Kernel::Clock::now();
+	}
+}
+
+void GameSpeed::stop()
+{
+	_running = false;
+	_score	 = 0;
+	_ledManager.stop();
+}
+
+auto GameSpeed::isRunning() const -> bool
+{
+	return _running;
+}
+
+auto GameSpeed::checkForTimeout() -> bool
+{
+	auto roundDuration		  = static_cast<int>((rtos::Kernel::Clock::now() - _startTime).count());
+	auto time_still_available = roundDuration < _maximumDuration;
+	if (!time_still_available) {
+		_score = 0;
+	}
+
+	return time_still_available;
+}
```

Comment:
```suggestion
auto GameSpeed::isTimeout() -> bool
{
	auto roundDuration		  = static_cast<int>((rtos::Kernel::Clock::now() - _startTime).count());
	auto timeout = roundDuration >= _maximumDuration;
	if (timeout) {
		_score = 0;
	}

	return timeout;
}
```

On préfère les conditions positives de préference ça fonctionne pareil mais pas besoin de réfléchir à une information puis son contraire.